src/service.ts

```diff
@@ -331,6 +332,37 @@ export class Service {
     }
     return uri;
   }
+
+  static downloadProject(project: Project, uri?: string) {
+    const zipFile: JSZip = new JSZip();
+    let zipName: string = "download.zip";
+    if (!isUndefined(uri)) {
+      zipName = `${uri}.zip`;
+    }
+    function defineZip(file: File, folder?: any) {
+      if (file instanceof Directory) {
+        let folder: any;
+        if (!(file instanceof Project)) {
+          folder = zipFile.folder(file.name);
+        }
+        file.mapEachFile((file: File) => defineZip(file, folder));
+      } else {
+        if (!isUndefined(folder)) {
+          folder.file(file.name, file.data);
+        } else {
+          zipFile.file(file.name, file.data);
+        }
+      }
+    }
+    defineZip(project);
+    zipFile.generateAsync({type: "blob", mimeType: "application/zip"}).then((blob: any) => {
+      const link = document.createElement("a");
+      link.download = zipName;
+      link.href = URL.createObjectURL(blob);
+      link.click();
```

Comment:
it does not work on FF -- can you add `link` to the DOM and then remove it

Reply:
You mean by doing something like:
`document.body.appendChild(link); `
`link.click();`
`document.body.removeChild(document.body.lastChild);`

Right?